src/ldp/http/TextErrorHandler.ts

```diff
@@ -0,0 +1,30 @@
+import { getStatusCode } from '../../util/errors/ErrorUtil';
+import { guardedStreamFrom } from '../../util/StreamUtil';
+import { toLiteral } from '../../util/TermUtil';
+import { HTTP, XSD } from '../../util/Vocabularies';
+import { RepresentationMetadata } from '../representation/RepresentationMetadata';
+import type { ErrorHandlerArgs } from './ErrorHandler';
+import { ErrorHandler } from './ErrorHandler';
+import type { ResponseDescription } from './response/ResponseDescription';
+
+/**
+ * Returns a simple text description of an error.
+ * This class is mostly a failsafe in case all other solutions fail.
+ */
+export class TextErrorHandler extends ErrorHandler {
+  public async handle({ error }: ErrorHandlerArgs): Promise<ResponseDescription> {
+    const statusCode = getStatusCode(error);
+    const metadata = new RepresentationMetadata('text/plain');
+    metadata.add(HTTP.terms.statusCodeNumber, toLiteral(statusCode, XSD.terms.integer));
+
+    const text = typeof error.stack === 'string' ?
```

Comment:
Perhaps have a `showStackTrace` boolean constructor argument (for security reasons). Can default to `true`.

Reply:
Or actually, perhaps in `ConvertingErrorHandler` as `preserveStackTrace`?

Reply:
The `TextErrorHandler` is indepedent of what the `ConvertingErrorHandler` does so they both would need that constructor parameter. It's pretty much there as a backup in case the converting one fails. But giving both of them that constructor parameter does not seem like a bad idea.

Reply:
> The `TextErrorHandler` is indepedent of what the `ConvertingErrorHandler`

Ah really? Should we just have an ErrorToTextConverter then?

Reply:
That was the other possible solution, but there are some minor differences. The main reason I did it like this was to answer the question "what happens if the converter finds no match?". So this is a fallback for that situation. If we add it as a converter instead, it would have output type `text/plain`. But in case the request header had as type preferences `application/whatever` we would have no error handler to handle that case. This means the error would be thrown to back to the server factory, which I would to avoid as much as possible.

Besides that it seemed like a good idea to have a backup solution in case the conversion one failed for some reason since that is more complex code.

I just really wanted to make sure error serialization is always handled in an error handler when possible.

Reply:
Makes sense. I just don't like that the stack option would have to be specified twice.

So how about we make the `showStackTrace` default to `false`, but set it to `true` in our configs (twice)? Then at least it's explicit.

Reply:
(Although note that we do have the server factory fallback in any case. But yes, `application/whatever` shouldn't be that case.)

Reply:
Sounds good.

Ideally the server factory callback should never be reached, except when something goes wrong while writing a response.

Note that the server factory currently also outputs the stack in case it receives an error before headers are sent. So we would either also have to add the boolean, or simply disable stack printing completely there.

Reply:
Disable stack printing, for security reasons.

Reply:
For the other two, we could introduce a new CLI variable `showStackTrace` for this which is false by default.

Reply:
Great idea. In that case, we can even add the parameter to the server factory, it's just a single config.

Reply:
> For the other two, we could introduce a new CLI variable showStackTrace for this which is false by default.

Oh, good idea, I might steal that for Comunica as well :-)